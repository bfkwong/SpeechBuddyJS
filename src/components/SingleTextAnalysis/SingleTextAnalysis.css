.left-align {
  text-align: left !important;
}

.SingleTextAnalysis {
  background-color: #e0f5f5;
  color: #212c31;
  border-radius: 10px;
  padding: 20px 15px 15px 15px;
  margin: 10px 0px 0px 0px;
}

p {
  margin: 0px 0px 0px 0px !important;
}

.vertical-alignment {
  vertical-align: middle !important;
  height: 100%;
}

.SingleTxtAnalysis__h3 {
  padding-bottom: 20px;
}

.SingleTxtAnalysis__h5 {
  text-align: left;
}
