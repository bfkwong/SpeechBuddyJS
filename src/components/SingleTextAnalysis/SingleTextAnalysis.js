import React from "react";
import { Container, Table } from "react-bootstrap";
import { Radar, Bar } from "react-chartjs-2";
import "./SingleTextAnalysis.css";

function SingleTextAnalysis(props) {
  if (props.analysis.sentiment === undefined) {
    return <div></div>;
  }

  return (
    <Container className="SingleTextAnalysis">
      <h3 className="SingleTxtAnalysis__h3">Text Analysis</h3>

      <h5 className="SingleTxtAnalysis__h5">Top Ten Words</h5>
      <Bar
        data={{
          labels: props.analysis.topTenWords.map((v) => v[0]),
          datasets: [
            {
              label: `Top ${props.analysis.topTenWords.length} Words`,
              data: props.analysis.topTenWords.map((v) => v[1])
            }
          ]
        }}
        options={{
          scales: {
            yAxes: [
              {
                ticks: {
                  beginAtZero: true
                }
              }
            ]
          }
        }}></Bar>

      <h5 className="SingleTxtAnalysis__h5">Sentiment Analysis</h5>
      <Table striped bordered hover>
        <tbody>
          <tr>
            <th>Sentiment</th>
            <th>{props.analysis.sentiment}</th>
          </tr>
        </tbody>
      </Table>

      <h5 className="SingleTxtAnalysis__h5">Toxicity Analysis</h5>
      <Radar
        data={{
          labels: props.analysis.toxicity.map((t) => t.label),
          datasets: [
            {
              label: "Likelihood of Toxicity",
              backgroundColor: "#212C31",
              data: props.analysis.toxicity.map(
                (t) => t.results[0].probabilities[1]
              )
            }
          ]
        }}
        options={{
          scale: {
            ticks: {
              max: 1,
              min: 0,
              stepSize: 0.2
            }
          }
        }}
      />
    </Container>
  );
}

export default SingleTextAnalysis;
