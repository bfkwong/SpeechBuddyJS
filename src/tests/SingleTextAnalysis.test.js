import React from "react";
import { render } from "@testing-library/react";
import SingleTextAnalysis from "../components/SingleTextAnalysis/SingleTextAnalysis";
import renderer from "react-test-renderer";
import { unmountComponentAtNode } from "react-dom";

describe("SingleTextAnalysis Component", () => {
  let container;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    if (container.hasChildNodes()) {
      unmountComponentAtNode(container);
    }
    container.remove();
    container = null;
  });

  test("SingleTextAnalysisComponent renders without failure", () => {
    render(<SingleTextAnalysis />, container);
  });

  test("SingleTextAnalysisComponent snapshot test", () => {
    const appTree = renderer.create(<SingleTextAnalysis />).toJSON();
    expect(appTree).toMatchSnapshot();
  });
});
