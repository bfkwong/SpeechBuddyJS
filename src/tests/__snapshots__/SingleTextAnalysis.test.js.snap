// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleTextAnalysis Component SingleTextAnalysisComponent snapshot test 1`] = `<div />`;
